File: tcpcom/PC/Python/Examples/ButtonClient.py
# ButtonClient.py


from tcpcom import TCPClient
from easygui import msgbox, enterbox

def onStateChanged(state, msg):
    print "State: " + state + ". Message: " + msg

port = 5000 # IP port
host= enterbox("Button Server IP Address?", "Button Client", "", True)
if host != None:
    client = TCPClient(host, port, stateChanged = onStateChanged)
    rc = client.connect()
    if rc:
        msgbox("Button client running. OK to stop","Button Client")
        client.disconnect()

File: tcpcom/PC/Java/Examples/ButtonClient.java
import ch.aplu.tcpcom.*;
import ch.aplu.util.*;

public class ButtonClient implements TCPClientListener
{
  private ModelessOptionPane mop;

  public ButtonClient()
  {
    int port = 5000;
    InputDialog id = new InputDialog("Button Client", "Host?");
    String host = id.readString();
    if (host == null)
      return;
    TCPClient client = new TCPClient(host, port);
    boolean success = client.connect();
    if (!success)
    {
       mop = new ModelessOptionPane("Connection failed");
       return;
    }
    mop = new ModelessOptionPane("Information from server:");
    mop.setTitle("Button Client");
    client.addTCPClientListener(this);
  }

  public void onStateChanged(String state, String msg)
  {
     mop.setText("State: " + state + ". Message: " + msg);
  }
  
  public static void main(String[] args)
  {
    new ButtonClient();
  }
}